reaction_smi

juditzador opened the pull request:
This is the function that prints the smiles of each reaction along with the relevant energies.

Commit:
reaction_smi

Commit:
fixed error with duplicate products for reaction_smi

rubenvdvijver commented on pesviewer/pesviewer.py:
```diff
@@ -910,9 +910,50 @@ def get_smis(m, smis, files):
                 # end try
             # end for
-            return smis
+        return(smis)
```
I don't understand why you changed the indentation and added brackets here.

rubenvdvijver commented on pesviewer/pesviewer.py:
```diff
@@ -910,9 +910,50 @@ def get_smis(m, smis, files):
+        The barrier height for barrierless reactions is given as max(Ereact,Eprod)
+        """
+
+        with open('reaction_smi.out', 'w') as f:
```
Could it sometimes happen that two pesviewers are called in the same directory? If so, the reaction_smi.out will be overwritten. That is why I add the `options['id']` to all file names.

rubenvdvijver commented on pesviewer/pesviewer.py:
```diff
@@ -910,9 +910,50 @@ def get_smis(m, smis, files):
     # end def
 
+    def reaction_smi():
+        """
```
I am thinking whether this is the best place to print this file, this is in the generate_2d_depiction method, which has not much to do with the new feature. I can't see another method that would fit better.

rubenvdvijver reviewed (commented):
Thanks for the feature implementation. Do you have some other codes to postprocess this reaction_smi.out file or is it just to help you analyze the pes?

RubenVandeVijver committed:
Resolved bug in case pybel and rdkit are both unavailable.

RubenVandeVijver committed:
Added the long description to the setup.py

RubenVandeVijver committed:
Added conda files

RubenVandeVijver committed:
Added Pillow to required modules in the setup file

RubenVandeVijver committed:
Added the options of having vertical lines instead of polynomials between the stationary points. Also added the import for OpenBabel3.0.0.

RubenVandeVijver committed:
Fixed merge conflict

RubenVandeVijver committed:
Using linear lines is an option in the input file

rubenvdvijver committed:
Update README.md

add linear_lines as input option to the list of options

juditzador committed:
added interpolation option, hanning as default, to make nicer pictures

juditzador committed:
Merge branch 'master' of github.com:/zadorlab/PESViewer

juditzador committed:
solved the problem of skewing structures when the whole picture gets larger

juditzador commented:
Hi Ruben,
I didn't react to your old questions about printing reaction smiles, feel free to revert those changes if you don't think that they are useful. However, I fixed two things. The molecules are printed much nicer with the hanning interpolation, and now if you move the stationary points outside the original range (or make the range smaller), the structures won't get fat or skinny.

juditzador committed:
added graph depiction

juditzador committed:
small changes

juditzador committed:
updated input.txt with graph edge color

juditzador committed:
Update README.md

juditzador committed:
uncommented pes plot

juditzador committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

juditzador committed:
edges are scaled in width and using a better color scale

juditzador committed:
fixed line width problem

juditzador committed:
fixed line color, but not yet perfect

cmartia committed:
Improved image quality of 2D chemical structures

cmartia committed:
Changes to improve the quality of the 2D depictions.

cmartia committed:
Further fine tuning of drawing parameters.

cmartia committed:
Module to generate all possible resonant structures of a given molecule.

cmartia committed:
Minor stylistic corrections

cmartia committed:
Draw depictions for each resonant structure. And further improed its quality for both old and new rdkit methods to draw 2D depicitons.

cmartia committed:
Improved docstrings and minor stylistic corrections.

cmartia committed:
Filter out resonant structures already been generated.

cmartia committed:
Renaming of resonant structures.

cmartia committed:
Do not generate images if already present.

cmartia committed:
Adapt image size as per the molecule size,

cmartia committed:
Minor readability improvements.

cmartia committed:
Correct bug when falling back to drawing structures with openbabel, configid was not defined.

cmartia committed:
Correct import of gen_resonant_structs. Added message when using openbabel as fallback option.

cmartia committed:
Correct import of gen_resonant_structs. Added message when using openbabel as fallback option.

juditzador committed:
slightly changed import syntax for gen_resonant_structs

juditzador committed:
added barrirless depiction to graph

juditzador committed:
added an fs parameter to options to scale fig

cmartia committed:
Corrected wrong smiles for some N-containing structures.

cmartia committed:
Larger size for energy labels

cmartia committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

juditzador committed:
small changes

juditzador committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

cmartia committed:
Added support for Sulphur and Halogens in the resonant structure generation including multivalence.

juditzador committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

cmartia committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

cmartia committed:
Discard structures with hipervalent C not catched by Chem.SanitizeMol() before the assignement of the radical sites.

juditzador committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

cmartia committed:
Allows inline comments (starting with #) in the input file.

cmartia committed:
Fixed the final TypeError message at the end of the pesviewer execution when run from the entry point.

cmartia committed:
Removed openbabel warnings to be printed. Minor stylistic corrections.

cmartia committed:
Minor stylistic changes.

cmartia committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

 Conflicts:
	pesviewer/pesviewer.py

juditzador committed:
interactive for barrierless uses product energy

juditzador committed:
more barrierless stuff

juditzador committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

juditzador committed:
fixed barrierless channel depiction in the graph

juditzador committed:
different color for bimoleculars

cmartia committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

juditzador committed:
removed unnecessary print

juditzador committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

juditzador committed:
extended pubel export and added message if import fails

juditzador committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

cmartia committed:
Allow the usage of smiles instead of xyz files to generate 2d depictions.

cmartia committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

cmartia committed:
Raise an error when no rdkit mol object is generated.

cmartia committed:
Allow structures with weird valences (eg. N: 4, 5).

cmartia committed:
Allow to enable/disable the generation of resonant structures for the molecular 2d depictions.

cmartia committed:
Set the resolution and linewidth of the traditional Pot. vs Reac. Coord. plot from the input file.

juditzador committed:
Update README.md

juditzador committed:
Update README.md

juditzador committed:
Update README.md

juditzador committed:
Update README.md

juditzador committed:
Update README.md

juditzador committed:
Update README.md

juditzador committed:
Update README.md

juditzador committed:
Update README.md

juditzador committed:
Update README.md

cmartia committed:
Add a .gitignore file.

juditzador committed:
can find bottle neck between two species

juditzador committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

cmartia committed:
Added highlight color to graph representation.

cmartia committed:
Also highlight the node contour.

juditzador committed:
making input for lowest E path plot automatically

juditzador committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

juditzador committed:
small bug

juditzador committed:
shortest of equal paths is chosen for minE path

juditzador committed:
fixed file name

juditzador committed:
exclude connections via bimoleculars

cmartia committed:
Added energy attribute to the barrierless class.

It takes the value of the self.product.energy.

cmartia committed:
Function to generate a networkx graph of the PES.

cmartia committed:
Added .vscode dir in .gitignore.

cmartia committed:
Minor stylistic details.

cmartia committed:
Allow to carry out several MEP searches.

Also carry out a general refactoring of an important part of the code.

cmartia committed:
Improved docstrings.

cmartia committed:
Color MEPs in the graph representation of the PES.

cmartia committed:
Correct the color of bimolecular and wells.

cmartia committed:
Corrected bug when rescale was not set.

cmartia committed:
Change the default color of barrrierless to gray.

cmartia committed:
Write instructions to create MEP 2D plots.

Instead of creating plots for each of the MEPs requested, give
instructions on how to do it.

cmartia committed:
Allow nodes to have a proportional size to their stability.

cmartia committed:
Allow the user to change the size difference of the nodes.

Change the input parameter prop_node_size to node_size_diff.

cmartia committed:
Updated README.md with latest options.

cmartia committed:
Reorder option to fit documentation.

cmartia committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

juditzador committed:
Update README.md

cmartia committed:
Added .DS_store in the .gitconfig.

cmartia committed:
Make the canvas around each 2d depiction transparent.

juditzador committed:
added display_units keyword to convert units on display

juditzador committed:
units fixed to display_units when graph mode is used

juditzador committed:
Update README.md

juditzador committed:
with display_units the Y scaling is fixed

juditzador committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

juditzador committed:
fixed Y range

juditzador committed:
xx

juditzador committed:
found and fixed bug for yhigh

cmartia committed:
Resorder import statements.

cmartia committed:
Allow pesviewer to be imported as a library.

cmartia committed:
Added wrapper function around main.

cmartia committed:
Make all dependencies mandatory.

cmartia committed:
Updated installation procedure

cmartia committed:
Added example input inside examples.

cmartia committed:
Updated meta.yaml to make a conda-forge package.

cmartia committed:
Added further details in the conda recipe.

cmartia committed:
Updated version and added wiki url.

cmartia committed:
Remove openbabel and rdkit from the pip requirements.

cmartia committed:
Added updated meta.yaml for v1.1.0

cmartia committed:
Added white background to nodes in the graph to compensate for transparent canvas.

cmartia committed:
On the graph only highligh the border of nodes.

cmartia committed:
Disable the generation of all generation structures by default

cmartia committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

juditzador committed:
added option for significan digits called rounding

juditzador committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

cmartia committed:
Updated installation instructions

juditzador committed:
adjusted rounding option for traditional PES

juditzador committed:
added rounding

juditzador committed:
implemented 'rounding' and 'energy_shift' params

juditzador committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

juditzador committed:
Update README.md

cmartia committed:
Update .gitignore.

juditzador committed:
fixed printing bug for path search

juditzador committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

cmartia committed:
Allow storing a second energy in wells and bimoleculars.

cmartia committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

cmartia committed:
Use second energy when the first one is nan.

cmartia committed:
When two energies are provided write both values as labels of the graph.

cmartia committed:
Allow defining color of bimolecular species border in the graph, from the input..

cmartia committed:
Get rid of energy-dependent node sizes in the graph and correct the labels when several energies are present.

cmartia committed:
Remove arrows from the graph.

carles-marti committed:
Correct bug in resonant structures.

cmartia committed:
Add Xenon to elements in reso_2d.

cmartia committed:
Implemented parameter to able/disable the plot of the graph.

cmartia committed:
Correct radical sites in charged species.

cmartia committed:
Add the possibility to make gray dotted lines whem color is set to dotted.

juditzador committed:
do not make png for vdW wells

juditzador committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

juditzador committed:
sensible default for save

juditzador committed:
Merge branch 'master' of github.com:zadorlab/PESViewer

juditzador committed:
new release

juditzador committed:
update hash

juditzador committed:
printing smiles

juditzador committed:
Merge branch 'master' of github.com:zadorlab/PESViewer
